js/layout-javascript/news-feed-code.js

```diff
@@ -2530,48 +2600,105 @@ DynamicList.prototype.deleteComment = function(id) {
+  var options = {
+    instance: _this,
+    config: _this.data,
+    id: _this.data.id,
+    uuid: _this.data.uuid,
+    container: _this.$container,
+    record: entry,
+    commentId: id,
+    commentContainer: commentHolder
+  };
 
-  return Fliplet.DataSources.connect(_this.data.commentsDataSourceId).then(function (connection) {
-    return connection.removeById(id, { ack: true });
-  }).then(function onRemove() {
-    _.remove(entry.comments, { id: id });
-    entry.commentCount--;
-    _this.updateCommentCounter({
-      id: entryId,
-      record: entry
+  commentHolder.hide();
+
+  return Fliplet.Hooks.run('flListDataBeforeDeleteComment', options).then(function () {
+    return Fliplet.DataSources.connect(_this.data.commentsDataSourceId).then(function (connection) {
+      return connection.removeById(id, { ack: true });
+    }).then(function onRemove() {
+      _.remove(entry.comments, { id: id });
+      entry.commentCount--;
+      _this.updateCommentCounter({
+        id: entryId,
+        record: entry
+      });
+      commentHolder.remove();
+      Fliplet.Hooks.run('flListDataAfterDeleteComment', options);
+    });
+  }).catch(function (error) {
+    commentHolder.show();
+    Fliplet.UI.Toast.error(error, {
+      message: 'Error deleting comment'
     });
-    commentHolder.remove();
   });
 }
 
-DynamicList.prototype.saveComment = function(entryId, commentId, value) {
+DynamicList.prototype.saveComment = function(entryId, commentId, newComment) {
   var _this = this;
   var entry = _.find(_this.listItems, { id: entryId });
   var entryComments = _.get(entry, 'comments', []);
-  var commentData = _.find(entryComments.entries, function(comment) {
-    return comment.id === commentId;
-  });
+  var commentData = _.find(entryComments, { id: commentId });
 
-  if (entryComments) {
-    commentData = _.find(entryComments, function(comment) {
-      return comment.id === commentId;
-    });
+  if (!commentData) {
+    return Promise.reject('Comment not found');
   }
 
-  if (commentData) {
-    commentData.data.settings.text = value;
-    _this.replaceComment(commentId, commentData, 'temp');
-  }
+  var oldCommentData = _.clone(commentData);
+  var options = {
+    instance: _this,
+    config: _this.data,
+    id: _this.data.id,
+    uuid: _this.data.uuid,
+    container: _this.$container,
+    record: entry,
+    oldCommentData: oldCommentData,
+    newComment: newComment
+  };
 
-  return Fliplet.Content({ dataSourceId: _this.data.commentsDataSourceId })
-    .then(function(instance) {
-      return instance.update({
-        settings: commentData.data.settings
-      }, {
-        id: commentId
-      });
+  return Fliplet.Hooks.run('flListDataBeforeUpdateComment', options)
+    .then(function () {
+      newComment = options.newComment;
+
+      if (!newComment) {
+        return Promise.resolve();
+      }
+
+      commentData.data.settings.text = newComment;
+      _this.replaceComment(commentId, commentData, 'temp');
```

Comment:
the word `temp`/`tmp` is forbidden, please use an alternative to give more context

Reply:
Should I refactor old code. This isn't new code from me.